pkg/sdn/api/validation/validation.go

```diff
@@ -40,12 +40,37 @@ func ValidateClusterNetwork(clusterNet *sdnapi.ClusterNetwork) fielderrors.Valid
 	return allErrs
 }
 
+func validateNewNetwork(obj *sdnapi.ClusterNetwork, old *sdnapi.ClusterNetwork) error {
+	oldBase, oldNet, err := net.ParseCIDR(old.Network)
+	if err != nil {
+		// Shouldn't happen, but if the existing value is invalid, then any change should be an improvement...
+		return nil
+	}
+	oldSize, _ := oldNet.Mask.Size()
+	_, newNet, err := net.ParseCIDR(obj.Network)
+	if err != nil {
+		return fielderrors.NewFieldInvalid("Network", obj.Network, err.Error())
+	}
+	newSize, _ := newNet.Mask.Size()
+	if newSize < oldSize && newNet.Contains(oldBase) {
```

Comment:
I dislike 'size' since this was making thing "Why would we only allow it if the size of the network was smaller?"

Of course this is actually the number of leading 1s.  Can we call it oldOnes and newOnes? So some future idiot me doesn't get confused?  but LGTM even the way it is if you don't want to change it.